configuration/bahmniforms/history_and_examination_form.json

```diff
@@ -0,0 +1,47 @@
\",\"label\":{\"translationKey\":\"CONSULTATION_VIDEOS_11\",\"type\":\"label\",\"value\":\"Consultation Videos\",\"id\":\"11\"},\"properties\":{\"addMore\":false,\"location\":{\"column\":0,\"row\":4}},\"id\":\"11\",\"concept\":{\"name\":\"Consultation Videos\",\"uuid\":\"9bb0795c-4ff0-0305-1990-000000000006\",\"datatype\":\"N/A\",\"conceptClass\":\"Misc\",\"conceptHandler\":null,\"set\":true,\"setMembers\":[{\"name\":\"Patient Video\",\"uuid\":\"eef8e969-269d-40c4-b7c9-182da11c2653\",\"datatype\":\"Complex\",\"conceptClass\":\"Video\",\"conceptHandler\":\"VideoUrlHandler\",\"units\":null,\"hiNormal\":null,\"lowNormal\":null,\"hiAbsolute\":null,\"lowAbsolute\":null,\"answers\":[],\"properties\":{\"allowDecimal\":null}}]},\"controls\":[{\"type\":\"obsControl\",\"label\":{\"translationKey\":\"PATIENT_VIDEO_12\",\"id\":\"12\",\"units\":\"\",\"type\":\"label\",\"value\":\"Patient Video\"},\"properties\":{\"mandatory\":false,\"notes\":true,\"addMore\":true,\"hideLabel\":false,\"controlEvent\":false,\"location\":{\"row\":0,\"column\":0}},\"id\":\"12\",\"concept\":{\"name\":\"Patient Video\",\"uuid\":\"eef8e969-269d-40c4-b7c9-182da11c2653\",\"datatype\":\"Complex\",\"conceptClass\":\"Video\",\"conceptHandler\":\"VideoUrlHandler\",\"units\":null,\"hiNormal\":null,\"lowNormal\":null,\"hiAbsolute\":null,\"lowAbsolute\":null,\"answers\":[],\"properties\":{\"allowDecimal\":null}},\"units\":null,\"hiNormal\":null,\"lowNormal\":null,\"hiAbsolute\":null,\"lowAbsolute\":null}]}],\"events\":{\"onFormSave\":\"function(form) {\\n  let addmoreCount = 0;\\n  while (form.get('Chief Complaint Coded SNOMED', addmoreCount).currentRecord) {\\n    if (form.get('Chief Complaint Coded SNOMED', addmoreCount).getValue()) {\\n      if (!(form.get('Sign/symptom duration SNOMED', addmoreCount).getValue() && form.get('Chief Complaint Duration SNOMED', addmoreCount).getValue())) {\\n        throw {\\n          message: 'Please enter data in Chief complaint data'\\n        };\\n      } else if (form.get('Chief Complaint Coded SNOMED', addmoreCount).getValue() === 'Other generic') {\\n        if (!form.get('Chief complaint (text) SNOMED', addmoreCount).getValue()) {\\n          throw {\\n            message: 'Please enter Chief complaint (text)'\\n          };\\n        }\\n      }\\n    } else if ((form.get('Sign/symptom duration SNOMED', addmoreCount).getValue() || form.get('Chief Complaint Duration SNOMED', addmoreCount).getValue())) {\\n      throw {\\n        message: 'Please enter Chief complaint'\\n      };\\n    }\\n    addmoreCount++;\\n  }\\n}\",\"onFormInit\":\"function(form) {\\n  if (form.get('Chief Complaint Coded SNOMED').getValue() !== 'Other generic') {\\n    form.get('Chief complaint (text) SNOMED').setHidden(true);\\n  } else {\\n    form.get('Chief complaint (text) SNOMED').setHidden(false);\\n  }\\n}\"},\"translationsUrl\":\"/openmrs/ws/rest/v1/bahmniie/form/translations\",\"referenceVersion\":\"3\",\"referenceFormUuid\":\"9846c2d1-f4ca-44ff-a40b-b87a45b87267\"}",
```

Comment:
the form definition contains concept UUIDs that are non-standard (unlike say a CIEL concept). these concepts I think are created through the CSV import and therefore I am not sure if you can still have such references working in all env. They maybe working with the base CIEL loaded database Bahmni ships with.

Reply:
maybe I wasn't clear. are these specific to any particular database? for example, if you load through initializer CSVs, and UUIDs are not mentioned, then the system will generate new UUIDs. or if you are specifying the concept UUIDs that match a particular base database (Bahmni Lite comes with a preloaded db as well optionally), then for anybody starting with an empty database  - the above definition will not work.

Reply:
Ok. I think its ok, considering this is just a reference example. its better to describe in the readme that if a different base database or empty database is used, the example forms may not work.